src/LayerPlan.cpp

```diff
@@ -1124,6 +1125,65 @@ void LayerPlan::addLinesByOptimizer(const Polygons& polygons, const GCodePathCon
     }
 }
 
+void LayerPlan::addLinesMonotonic(const Polygons& polygons, const GCodePathConfig& config, const SpaceFillType space_fill_type, const AngleRadians monotonic_direction, const coord_t max_adjacent_distance, const coord_t wipe_dist, const Ratio flow_ratio, const double fan_speed)
+{
+    const Point last_position = getLastPlannedPositionOrStartingPosition();
+    PathOrderMonotonic<ConstPolygonRef> order(monotonic_direction, max_adjacent_distance, last_position);
+    for(size_t line_idx = 0; line_idx < polygons.size(); ++line_idx)
+    {
+        order.addPolyline(polygons[line_idx]);
+    }
+    order.optimize();
```

Comment:
Indeed this is the way it'll need to be called in Arachne, so we can unify them there.

The difficulty of unifying them now lies in that the result of the optimisation needs to be obtained differently. This makes the loop condition different (because the size is in the `paths` member, not in `polyOrder`) and also the method to obtain the actual paths. The old optimizer returned a list of indices which then need to be looked up in the original array. The new method contains the actual data in its optimised order. So in the new method the path is obtained with `order.paths[order_idx].vertices` while in the old method it is obtained with `polygons[orderOptimizer.polyOrder[order_idx]]`. This is done in two places, one nested very deeply, so it's hard to pull out.

One could write an adapter to convert one API to be the same as the other. But I don't think that's worth the effort considering it needs to be removed for Arachne anyway. The adapter is likely to give us more maintenance than this repetition, if we need to make a change here, too.

Opinions welcome of course. If you disagree I could still work on an adapter.

Reply:
I'm fine with postponing it until Arachne is in. My worry was just about it being permanent.